### PR TITLE
Add Symfony 7 support ; drop Symfony < 5.4 support and PHP < 8 support

### PR DESCRIPTION
**Pull request by VincentLanglet**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Fixed tickets | #257 #234 #256 #235 #251 #252
| License       | MIT

Hi @lsmith77, I know this bundle might be in low-maintenance mode. But adding SF 7 support would be useful:
- I dropped PHP 7 support and SF 4 support since they are not maintained.
- I added SF 7 support
- I update Php-cs-fixer dependency
- I fixed the tests and solved the phpunit deprecations

### CODE REVIEW

**Comment by CondonesMix** [comment]
sorry for the question.

It seems to be solved but composer require friendsofsymfony/ckeditor-bundle keeps failing me, has it already been published in the main branch?

Salutations from Barcelona

**Comment by VincentLanglet** [comment]
> It seems to be solved but composer require friendsofsymfony/ckeditor-bundle keeps failing me, has it already been published in the main branch?

That's the whole difference between a merged PR and an open PR.

**Comment by CondonesMix** [comment]
And is there a way to install a PR that is open like this?

**Comment by jcglombard** [comment]
@VincentLanglet Thx for this 👍🏾
@coreTeam please merge this PR to let us upgrade to SF7 😉

**Comment by avtarsm** [comment]
Please merge this PR.

File: composer.json [hunk 1]
```diff
@@ -17,33 +17,33 @@
...
-        "symfony/expression-language": "^4.4 || ^5.0 || ^6.0",
-        "symfony/form": "^4.4 || ^5.0 || ^6.0",
-        "symfony/framework-bundle": "^4.4 || ^5.0 || ^6.0",
-        "symfony/http-foundation": "^4.4 || ^5.0 || ^6.0",
-        "symfony/http-kernel": "^4.4 || ^5.0 || ^6.0",
-        "symfony/options-resolver": "^4.4 || ^5.0 || ^6.0",
-        "symfony/property-access": "^4.4 || ^5.0 || ^6.0",
-        "symfony/routing": "^4.4 || ^5.0 || ^6.0",
-        "symfony/twig-bundle": "^4.4 || ^5.0 || ^6.0",
+        "symfony/asset": "^5.4 || ^6.0 || ^7.0",
```
Comment (evertharmeling) [inline composer.json:23]: Maybe also bump the `^6.0` to `^6.3` at least (or even `^6.4`), as `<6.3` is unmaintained and `6.3` support ends this month... https://symfony.com/releases#symfony-releases-calendar

**Review by evertharmeling — APPROVED** [review]

File: composer.json (hunk 1 above)
Comment (VincentLanglet) [reply composer.json:23]: Previously the support was still `^5.0 || ^6.0",` even if it was unmaintained so I kept the same strategy ; even if I agree with you.

**Comment by Jean-Gian** [comment]
Great PR!
Do you guys think it will be merged and released soon?

**Comment by lsmith77** [comment]
I have commit access but I never used the bundle and I am not really using Symfony, so I don't feel comfortable tagging a release. I have tried to find someone with commit access to take this on ..

**Comment by Jean-Gian** [comment]
Do you think this project sort of abandoned?
I see this ultra simple PR created over 6 months ago that hasn't been merged https://github.com/FriendsOfSymfony/FOSCKEditorBundle/pull/250
It seems there is no activity nor interest by who can merge this

**Comment by mhpcc** [comment]
The project is definitely abandoned and there's little point in resurrecting it, as it uses CKEditor 4 which has been EOL for a while too. It would be great to get Symfony 7 support though, as it's not trivial to upgrade to CKEditor 5 due to changed licensing.

**Comment by Jean-Gian** [comment]
@lsmith77 considering what @mhpcc said, would you mind merging and tagging anyway?

**Comment by lsmith77** [comment]
ok .. from looking over the changes, they are all related to CS fixer. so safe enough.

**Comment by lsmith77** [comment]
@VincentLanglet did https://github.com/FriendsOfSymfony/FOSCKEditorBundle/pull/235 really get fixed with this PR?

**Comment by lsmith77** [comment]
OK this is now merged. I will wait a few more days for feedback until I will tag a release.

**Comment by Jean-Gian** [comment]
Amazing @lsmith77 ! Thanks a lot!
And thanks again @VincentLanglet for the PR

**Comment by mbabker** [comment]
> The project is definitely abandoned and there's little point in resurrecting it, as it uses CKEditor 4 which has been EOL for a while too. It would be great to get Symfony 7 support though, as it's not trivial to upgrade to CKEditor 5 due to changed licensing.

@mhpcc That's why I've opened #254 so that the current state of things can be better signposted.  This bundle works well enough and given how the last few releases are predominantly Symfony compat related, it clearly doesn't need a lot of maintenance.  But, at this point given the state of CKEditor 4, new users installing this bundle also isn't a great recommendation anymore.

**Comment by VincentLanglet** [comment]
> @VincentLanglet did #235 really get fixed with this PR?

Reading again the PR, I'm not sure. We can let it open, just in case...

**Comment by Gwadas** [comment]
for me is not work .. 
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - friendsofsymfony/ckeditor-bundle[1.0.0, ..., 1.2.1] require php ^5.6 || ^7.0 -> your php version (8.3.1) does not satisfy that requirement.
    - friendsofsymfony/ckeditor-bundle[2.0.0, ..., 2.2.0] require php ^7.1 -> your php version (8.3.1) does not satisfy that requirement.
    - friendsofsymfony/ckeditor-bundle 2.3.0 requires symfony/asset ^4.4 || ^5.0 -> found symfony/asset[v4.4.0, ..., v4.4.46, v5.0.0, ..., v5.4.31] but it conflicts with your root composer.json require (^7.0).
    - friendsofsymfony/ckeditor-bundle 2.4.0 requires symfony/asset ^4.4 || ^5.0 || ^6.0 -> found symfony/asset[v4.4.0, ..., v4.4.46, v5.0.0, ..., v5.4.31, v6.0.0, ..., v6.4.0] but it conflicts with your root composer.json require (^7.0).
    - Root composer.json requires friendsofsymfony/ckeditor-bundle * -> satisfiable by friendsofsymfony/ckeditor-bundle[1.0.0, 1.1.0, 1.2.0, 1.2.1, 2.0.0, ..., 2.4.0].

Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals

**Comment by lsmith77** [comment]
since I have not tagged a release, you need to allow dev dependencies

**Comment by VincentLanglet** [comment]
Or the version `dev-2.x as 2.5`

**Comment by lsmith77** [comment]
alright. time to tag a release.

**Comment by lsmith77** [comment]
https://github.com/FriendsOfSymfony/FOSCKEditorBundle/releases/tag/2.5.0

**Pull request merged**